src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -39,12 +42,85 @@ import {
   defaultDataSourceSelection,
 } from '../service/data_source_selection_service';
 import { DataSourceError } from '../types';
+import { DATACONNECTIONS_BASE } from '../constants';
+
+export const getDirectQueryConnections = async (dataSourceId: string, http: HttpSetup) => {
+  const endpoint = `${DATACONNECTIONS_BASE}/dataSourceMDSId=${dataSourceId}`;
+  const res = await http.get(endpoint);
+  const directQueryConnections: DataSourceTableItem[] = res.map(
```

Comment:
This should check `Array.isArray(res)` and probably throw and error if the response is not in the format we need.